### PR TITLE
Use clap to parse command arguments

### PR DESCRIPTION
**Pull request by DrChat**
Makes for a nicer end-user experience :)

```
kd> !snapshot --help
[snapshot] Usage: snapshot <KIND> [STATE_PATH]

Arguments:
  <KIND>        The kind of snapshot to take [possible values: active-kernel, full]
  [STATE_PATH]  The path to save the snapshot to

Options:
  -h, --help  Print help
```

### CODE REVIEW

**Comment by 0vercl0k** [comment]
This is awesome - thank you for sending this over! I just made nits changes - let me know if they look good to you and will merge this in :)

Also, TIL the `#[default]` attribute on the `Enum` which is super handy; cheers for sharing!

Cheers

File: src/lib.rs [hunk 1]
```diff
@@ -419,9 +431,19 @@ fn snapshot_with_kind(raw_client: RawIUnknown, kind: SnapshotKind, args: PCSTR)
         return E_ABORT;
     };
 
-    match snapshot_with_kind_inner(kind, &dbg, args) {
+    // Parse the arguments using `clap`. Currently splitting arguments by
+    // whitespaces. whitespace.
```
Comment (DrChat) [inline src/lib.rs]: extra bit left over from original comment here

**Comment by DrChat** [comment]
Thanks! And thanks for making this, this is great.
I'll probably look to push some more functionality in future PRs like wrappers for more WinDbg interfaces.

File: src/lib.rs (hunk 1 above)
Comment (0vercl0k) [reply src/lib.rs]: good catch 🤦🏼

**Comment by DrChat** [comment]
I'm good with the changes btw - feel free to send it :)

**Comment by 0vercl0k** [comment]
Awesome - thanks for taking a look! And yes happy to merge improvements in 🚀 

Cheers

**Pull request merged**